rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,422 @@
+            source=HistoricalPriceOracle.ALCHEMY,
+        )):
+            return price_cache_entry.price
+
+        try:
+            # if it's a crypto asset use symbol, otherwise use
+            asset_identifier = self._get_asset_identifier(from_asset)
+        except UnsupportedAsset as e:
+            log.warning(
+                f'Tried to query alchemy historical price from {from_asset.identifier} '
+                f'to {to_asset.identifier}. But from_asset is not supported in alchemy',
+            )
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+            ) from e
+
+        payload: dict[str, str | int] = {
+            'startTime': (start_of_day_timestamp := Timestamp(timestamp - (timestamp % DAY_IN_SECONDS))),  # noqa: E501
+            'endTime': start_of_day_timestamp + DAY_IN_SECONDS,
+            'interval': '1h',
+        }
+        result = self._query(
+            verb='POST',
+            endpoint='historical',
+            payload={**payload, **asset_identifier},
+        )
+        if (usd_price := self._deserialize_historical_price(
+            result=result,
+            asset_identifier=asset_identifier,
+            from_asset=from_asset,
+            timestamp=timestamp,
+            to_asset=to_asset,
+        )) == ZERO_PRICE:
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+                rate_limited=False,
+            )
+
+        if to_asset != A_USD:
+            # We need to query intermediate price in this case.
+            to_asset_price = PriceHistorian().query_historical_price(
+                from_asset=A_USD,
+                to_asset=to_asset,
+                timestamp=timestamp,
+            )
+            price = Price(usd_price * to_asset_price)
+        else:
+            price = usd_price
+
+        GlobalDBHandler.add_historical_prices(entries=[HistoricalPrice(
+            from_asset=from_asset,
+            to_asset=to_asset,
+            source=HistoricalPriceOracle.ALCHEMY,
+            timestamp=timestamp,
+            price=price,
+        )])
+        return price
+
+    def can_query_history(
+            self,
+            from_asset: Asset,  # pylint: disable=unused-argument
+            to_asset: Asset,
+            timestamp: Timestamp,
+            seconds: int | None = None,
+    ) -> bool:
+        return not self.is_penalized()
+
+    @staticmethod
+    def _get_asset_identifier(asset: AssetWithOracles) -> dict[str, str]:
+        """
+        May raise:
+        - UnsupportedAsset
+        """
+        if asset.is_evm_token() is True:
+            asset = asset.resolve_to_evm_token()
+            if asset.chain_id == ChainID.POLYGON_POS:
+                network = 'polygon-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_ONE:
```

Comment:
can you check in a small example if using `match`  warns you when you don't check all the cases of an enum?

Reply:
```py
from enum import Enum

class MyEnum(Enum):
    A = 1
    B = 2


def is_a(x: MyEnum):
    match x:
        case MyEnum.A:
            return True

a = is_a(MyEnum.B)
print(f'{a=}'). # a=None
```